packages/terra-graphs-docs/src/example-datasets/popup.js

```diff
@@ -0,0 +1,291 @@
+};
+const createTrackItem = (pair, label, items) => {
+    const item = pair.append("g").classed("popup-item", true);
+    item.append("label").classed("popup-label", true).text(label);
+    const section = item.append("g").classed("section", true);
+    for (const value of Object.values(items)) {
+        const content = section.append("g").classed("content", true);
+        Object.entries(value).forEach(([subLabel, subItem]) => {
+            const subContent = content.append("g").classed("popup-item", true);
+            subContent
+              .append("label")
+              .classed("popup-label", true)
+              .text(
+                subLabel
+                  .replace(/([a-z0-9])([A-Z])/g, "$1 $2")
+                  .replace(/(\b[a-z](?!\s))/g, (x) => x.toUpperCase())
+              );
+            subContent.append("g").classed("popup-text", true).text(subItem);
+        });
+    }
+};
+const renderPopup = (fn) => {
+    const tip = document.querySelector("#tooltip");
+    const clickHandler = () => {
+        d3.select(tip).attr("style", "display:none;").selectAll("g").remove();
+        if (utils.isFunction(fn)) {
+            fn();
+        }
+        d3.select("#overlay").remove();
+        tip.removeEventListener("click", clickHandler);
+        window.removeEventListener("resize", clickHandler);
+    };
+    // Add new popup
+    d3.select("body")
+      .append("div", "#tooltip")
+      .attr("id", "overlay")
+      .classed("overlay", true)
+      .on("click", clickHandler);
+    // Position popup
+    return d3
+      .select("#tooltip")
+      .style("left", `${d3.event.pageX + 5}px`)
+      .style("top", `${d3.event.pageY + 5}px`);
+};
+const renderY2Popup = (fn) => {
+    const tip = document.querySelector("#tooltip");
+    const clickHandler = () => {
+        d3.select(tip).attr("style", "display:none;").selectAll("g").remove();
+        if (utils.isFunction(fn)) {
+            fn();
+        }
+        d3.select("#overlay").remove();
+        tip.removeEventListener("click", clickHandler);
+        window.removeEventListener("resize", clickHandler);
+    };
+    // Add new popup
+    d3.select("body")
+      .append("div", "#tooltip")
+      .attr("id", "overlay")
+      .classed("overlay", true)
+      .on("click", clickHandler);
+    // Position popup
+    return d3
+      .select("#tooltip")
+      .style("left", `${d3.event.pageX - 250}px`)
+      .style("top", `${d3.event.pageY + 5}px`);
+};
+const removeOldPopup = () => {
+    // Remove old popup
+    d3.select("#overlay").remove();
+    if(document.getElementsByClassName('initial-tooltip')[0]) {
+        document.getElementsByClassName('initial-tooltip')[0].className = 'tooltip';
+    }
+    d3.select("#tooltip").attr("style", "").selectAll("g").remove();
+};
+export const loadPopup = (onCloseCB, key, index, value) => {
+    removeOldPopup();
+    const path = renderPopup(onCloseCB);
+    const pair = path.append("g");
+    if (value.x) {
```

Comment:
we could de-structure at least the first level properties of the `value` object, then update those references within the code block

```suggestion
    const pair = path.append("g");
    const { x, y, high, mid, low } = value;
    if (x) {
```

Reply:
@yuderekyu **Question**:  Is this part of the coding standard in terra? if not, Since we can use the variables in value directly, is it necessary to have an extra line that will destructure.

Reply:
Fixed [here](https://github.com/cerner/terra-graphs/pull/120/commits/aca7d1771ccb5429c7cb0d49767f99e18d9d1bec#diff-394515a80c97db2e9f0d6a749aed6f25e56409461e21de3f1b197aa9bfddd341R93).